packages/plugin-page-view-tracking/src/page-view-tracking.ts

```diff
@@ -0,0 +1,75 @@
+import { IAmplitudeBrowser } from '@amplitude/analytics-browser';
+import {
+  BaseEvent,
+  BeforePlugin,
+  BrowserClient,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import { PluginCampaignTracker } from './plugin-campaign-tracker';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export class PageViewTrackingPlugin implements BeforePlugin {
+  name = 'page-view-tracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(
+    private readonly instance: BrowserClient | IAmplitudeBrowser,
+    private readonly options: PageTrackingBrowserOptions = {},
+  ) {}
+
+  async setup(config: BrowserConfig): Promise<undefined> {
```

Comment:
we can do implicit return if we do the following

```suggestion
  async setup(config: BrowserConfig): Promise<void> {
```